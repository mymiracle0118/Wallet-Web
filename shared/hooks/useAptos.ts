import create from 'zustand'
import { devtools, persist } from 'zustand/middleware'
import { Ed25519Keypair } from '@mysten/sui.js/keypairs/ed25519'
import { chromeSyncStorage } from '@portal/shared/utils/chromeSyncStorage'
import { getAptosWalletUsingSeed, getBalance } from '@portal/shared/services/aptosService'
import { ethers } from 'ethers'
import { toUTF8Array } from '../utils/utf8'

type APTOSWalletState = {
  /** APTOS network - create wallet */
  createAptosWallet: (mnemonics?: string) => Promise<any>
  /** APTOS network - Update balance */
  refreshBalance: (aptosType?: string) => Promise<void>
  _keypair?: Ed25519Keypair | null
  /** APTOS network - Wallet Address */
  address?: string
  /** APTOS network - privateKey Address */
  privateKey?: string
  /** APTOS network - Total Balance of wallet */
  balance: number
  /** APTOS network - All objects owned by wallet */
  objectsOwned?: Array<any>

  wallet?: ethers.Wallet
  encryptedPrivateKey: (string | number)[]
  encryptedWallet?: string
}

const initialState = {
  address: undefined,
  balance: 0,
  _keypair: undefined,
  privateKey: undefined,
  encryptedPrivateKey: [],
}

export const useAptos = create<APTOSWalletState>()(
  devtools(
    persist(
      (set, get) => ({
        ...initialState,

        createAptosWallet: async (mnemonics?: string) => {
          const wallet = await getAptosWalletUsingSeed(mnemonics)
          if (wallet) {
            const encryptedPrivateKey = toUTF8Array(wallet?.privateKey)
            set({ address: wallet?.address, encryptedPrivateKey })
          }

          const { refreshBalance } = get()
          await refreshBalance()
          return wallet
        },
        refreshBalance: async (aptosType?: string) => {
          const { address } = get()
          if (!address) return
          const balance = await getBalance(address, aptosType)
          set({ balance: balance })
        },
      }),
      {
        name: 'APTOS-storage',
        getStorage: () => chromeSyncStorage,
        partialize: (state) => Object.fromEntries(Object.entries(state).filter(([key]) => !['wallet'].includes(key))), // Prevents decrypted wallet from being persisted in storage
      }
    ),
    { name: 'APTOS - Wallet' }
  )
)
