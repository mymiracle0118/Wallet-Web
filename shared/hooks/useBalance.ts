import { useEffect, useMemo } from 'react'
import { useQuery } from '@tanstack/react-query'
import { BigNumber } from 'ethers'

import { NetworkFactory } from '@portal/shared/factory/network.factory'
import { NetworkToken } from '../utils/types'
import { useStore } from './useStore'

type BalanceResponse = {
  balance: BigNumber | number
  balanceFormatted: number
  isLoading: boolean
}
const { updateTokenBalance } = useStore.getState()
export const useBalance = (token: NetworkToken): BalanceResponse => {
  let balance, balanceFormatted

  const networkFactory = NetworkFactory.selectByNetworkId(token.shortName)
  if (!networkFactory) {
    return {
      balance: 0,
      balanceFormatted: 0,
      isLoading: false,
    }
  }

  const walletAddress = token.address
  const { data, isLoading } = useQuery(
    ['asset-balance', token.networkName, walletAddress, token.shortName],
    () => networkFactory.getTokenBalance(walletAddress as string, token),
    {
      refetchInterval: 60000,
      staleTime: 30000,
    }
  )

  balance = useMemo(() => {
    return data || BigNumber.from(0)
  }, [data])
  balanceFormatted = useMemo(() => networkFactory.getBalanceFormattedValue(data), [data]) as number

  useEffect(() => {
    // Update asset balance in wallet to keep running total
    if (balance && balanceFormatted) {
      updateTokenBalance(token.shortName, balance, balanceFormatted)
    }
  }, [balance, balanceFormatted])

  return {
    balance,
    balanceFormatted,
    isLoading,
  }
}
