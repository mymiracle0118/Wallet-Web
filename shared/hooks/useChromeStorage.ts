import { useCallback, useEffect, useState } from 'react'

export const storage = {
  get: (key: string, defaultValue: any, storageArea: Storage) => {
    const keyObj = defaultValue === undefined ? key : { [key]: defaultValue }
    return new Promise((resolve, reject) => {
      chrome.storage[storageArea].get(keyObj, (items: any) => {
        const error = chrome.runtime.lastError
        if (error) return reject(error)
        resolve(items[key])
      })
    })
  },
  set: (key: string, value: any, storageArea: Storage) => {
    return new Promise((resolve, reject) => {
      chrome.storage[storageArea].set({ [key]: value }, () => {
        const error = chrome.runtime.lastError
        error ? reject(error) : resolve()
      })
    })
  },
}

/**
 * Basic hook for storage
 * @param {string} key
 * @param {*} initialValue
 * @param {'local'|'sync'} storageArea
 * @returns {[*, function(*= any): void, boolean, string]}
 */

export default function useChromeStorage(key, initialValue, storageArea) {
  const [INITIAL_VALUE] = useState(() => {
    return typeof initialValue === 'function' ? initialValue() : initialValue
  })
  const [STORAGE_AREA] = useState(storageArea)
  const [state, setState] = useState(INITIAL_VALUE)
  const [isPersistent, setIsPersistent] = useState(true)
  const [error, setError] = useState('')

  useEffect(() => {
    storage
      .get(key, INITIAL_VALUE, STORAGE_AREA)
      .then((res) => {
        setState(res)
        setIsPersistent(true)
        setError('')
      })
      .catch((error) => {
        setIsPersistent(false)
        setError(error)
      })
  }, [key, INITIAL_VALUE, STORAGE_AREA])

  const updateValue = useCallback(
    (newValue) => {
      const toStore = typeof newValue === 'function' ? newValue(state) : newValue
      storage
        .set(key, toStore, STORAGE_AREA)
        .then(() => {
          setIsPersistent(true)
          setError('')
        })
        .catch((error) => {
          // set newValue to local state because chrome.storage.onChanged won't be fired in error case
          setState(toStore)
          setIsPersistent(false)
          setError(error)
        })
    },
    [STORAGE_AREA, key, state]
  )

  useEffect(() => {
    const onChange = (changes, areaName) => {
      if (areaName === STORAGE_AREA && key in changes) {
        setState(changes[key].newValue)
        setIsPersistent(true)
        setError('')
      }
    }
    chrome.storage.onChanged.addListener(onChange)
    return () => {
      chrome.storage.onChanged.removeListener(onChange)
    }
  }, [key, STORAGE_AREA])

  return [state, updateValue, isPersistent, error]
}
