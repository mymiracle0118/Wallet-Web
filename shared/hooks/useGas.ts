import React, { useEffect, useState } from 'react'
import { useQuery } from '@tanstack/react-query'
import { ethers } from 'ethers'
import { getContractInterface } from '@portal/shared/utils/getContractInterface'
import { NetworkFactory } from '@portal/shared/factory/network.factory'
import { Asset, GasOption, NetworkToken } from '@portal/shared/utils/types'

const getERC20GasLimit = async (asset: Asset, provider: any): Promise<number> => {
  if (asset.type === 'erc721') {
    return 100000
  }
  const contract = new ethers.Contract(asset.contractAddress!, getContractInterface(asset.type), provider)
  const deploymentData = contract.interface.encodeDeploy([])
  const estimatedGasLimit = await provider.estimateGas({ data: deploymentData })
  return Number(ethers.utils.formatUnits(estimatedGasLimit, 'wei'))
}

export const useGas = (asset: NetworkToken, option?: GasOption) => {
  const [gasLimit, setGasLimit] = useState(21000)
  const [gasOption, setGasOption] = useState(option || GasOption.Aggressive)
  const getNetwork = NetworkFactory.selectByNetworkId(asset.networkName)
  const provider = getNetwork.provider
  const { data: feeData } = useQuery(
    ['gas-fee-data', asset],
    () => {
      return getNetwork.getFeeData()
    },
    {
      staleTime: 15000,
      refetchInterval: 15000,
    }
  )

  const { data: gasLimitData } = useQuery(
    ['gas-limit-data', asset, provider],
    () => getERC20GasLimit(asset, provider),
    {
      enabled: asset.tokenContractAddress !== undefined,
      staleTime: 60 * 60000,
    }
  )

  useEffect(() => {
    if (gasLimitData) {
      setGasLimit(gasLimitData)
    }
  }, [gasLimitData])

  const maxFeePerGas = React.useMemo(() => {
    if (!feeData || feeData.gasPrice === null) {
      return ethers.utils.parseUnits('0', 'gwei')
    }

    let baseGas = parseFloat(ethers.utils.formatUnits(feeData.gasPrice || 0, 'gwei'))
    const baseGasPlusMultiplier = (multiplier: number) => {
      return baseGas * multiplier
    }

    switch (gasOption) {
      case GasOption.Low:
        baseGas = baseGasPlusMultiplier(1.0)
        break
      case GasOption.Market:
        baseGas = baseGasPlusMultiplier(1.5)
        break
      case GasOption.Aggressive:
        baseGas = baseGasPlusMultiplier(2.0)
        break
    }

    return ethers.utils.parseUnits(baseGas.toFixed(8), 'gwei')
  }, [gasOption, feeData])

  const maxPriorityFeePerGas = React.useMemo(() => {
    return getNetwork.getMaxPriorityFeePerGas({ maxFeePerGas, gasOption })
  }, [gasOption, maxFeePerGas])

  const marketGas = React.useMemo(async () => {
    return await getNetwork.marketGas(maxFeePerGas)
  }, [gasOption, maxFeePerGas])

  const estimatedTransactionCost = React.useMemo(async () => {
    return maxFeePerGas && (await getNetwork.getEstimatedTransactionCost(maxFeePerGas, maxPriorityFeePerGas, gasLimit))
  }, [gasLimit, gasOption, maxFeePerGas])

  const estimatedTransactionTime = React.useMemo(() => {
    switch (gasOption) {
      case GasOption.Low:
        return '30 seconds'
      case GasOption.Market:
        return '30 seconds'
      case GasOption.Aggressive:
        return '15 seconds'
    }
  }, [gasOption])

  return {
    gasLimit,
    gasOption,
    setGasOption,
    marketGas,
    maxFeePerGas,
    maxPriorityFeePerGas,
    estimatedTransactionCost,
    estimatedTransactionTime,
  }
}
