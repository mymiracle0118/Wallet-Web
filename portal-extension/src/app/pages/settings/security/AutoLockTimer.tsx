import React, { useState } from 'react'
import { useTranslation, Trans } from 'react-i18next'
import SinglePageTitleLayout from 'layouts/single-page-layout/SinglePageLayout'
import { Button, CustomTypography, Input } from 'components'
import { useWallet } from '@portal/shared/hooks/useWallet'
import { goBack } from 'lib/woozie'

const AutoLockTimer = () => {
  const { t } = useTranslation()
  const { lockTime, setLockTimer } = useWallet()
  const [minutes, setMinutes] = useState<string | number>(lockTime || 5)
  const [error, setError] = useState<boolean>(false)

  const handleTimeChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const value = event.target.value
    setMinutes(value as string)
    if (Number(value) > 120 || Number(value) < 1) {
      setError(true)
    } else {
      setError(false)
    }
  }

  const onSaveTime = () => {
    setLockTimer(Number(minutes))
    goBack()
  }

  return (
    <SinglePageTitleLayout dataAid="autoLockHead" title={t('Security.autoLockTimer')}>
      <div className="space-y-4">
        <CustomTypography variant="body">
          <Trans i18nkey="Security.autoLockInMins" values={{ inMins: 'in minutes' }}>
            Set the idle time <span className="font-extrabold">in minutes</span> before the wallet will be locked.
          </Trans>
        </CustomTypography>

        <div className="space-y-1 min-h-[4.0625rem]">
          <Input
            dataAid="inputTime"
            dataTestId="input-lock-time"
            type="number"
            fullWidth
            mainColor
            placeholder="1-120 minutes"
            value={minutes}
            icon={<CustomTypography variant="small">mins</CustomTypography>}
            onChange={handleTimeChange}
          />
          {error && <div className="text-xs text-feedback-negative">Set Minutes between 1-120</div>}
        </div>

        <div className="flex pt-4 justify-center gap-2">
          <Button data-aid="cancelButton" variant="bordered" color="outlined" onClick={goBack}>
            {t('Actions.cancel')}
          </Button>
          <Button
            data-aid="saveButton"
            data-test-id="button-save"
            color={`${Number(minutes) > 120 || lockTime === Number(minutes) ? 'disabled' : 'primary'}`}
            isDisabled={Number(minutes) > 120 || lockTime === Number(minutes)}
            onClick={onSaveTime}
          >
            {t('Actions.save')}
          </Button>
        </div>
      </div>
    </SinglePageTitleLayout>
  )
}

export default AutoLockTimer
