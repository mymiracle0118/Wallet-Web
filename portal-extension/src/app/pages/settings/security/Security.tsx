import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useNavigate } from 'lib/woozie'
import SinglePageTitleLayout from 'layouts/single-page-layout/SinglePageLayout'
import { Switch } from 'components'
import { useSettings } from '@portal/shared/hooks/useSettings'
import SettingItem from '../SettingItem'
import PasswordPromptModal from 'app/pages/wallet/PasswordPromptModal'
import { ArrowRight } from '@src/app/components/Icons'

const Security = () => {
  const { t } = useTranslation()
  const { navigate } = useNavigate()
  const { enablePasswordProtection, setEnablePasswordProtection } = useSettings()
  const [openPasswordModal, setOpenPasswordModal] = useState<boolean>(false)
  const [isHideAccountBlalance, setHideAccountBlalance] = useState<boolean>(false)

  const passwordPromptFailure = (error: Error) => {
    console.log('Wrong Password', error)
  }

  return (
    <SinglePageTitleLayout title="Security">
      <div className="border border-solid border-custom-white10 rounded-lg bg-surface-dark-alt">
        <SettingItem
          dataAid="secretRecoveryPhraseNavigation"
          title={t('Security.secretRecoveryPhrase')}
          endAndorment={<ArrowRight />}
          onClick={() => navigate('/settings/security/secret-recovery-phrase')}
        />
        <SettingItem
          dataAid="passwordProtectionPop"
          title={t('Security.passwordProtectionTitle')}
          subTitle={t('Security.passwordProtectionSubTitle') || ''}
          endAndorment={
            <Switch
              dataAid="passwordSwitch"
              id="switch-enable-password-protection"
              checked={enablePasswordProtection}
              onChange={(checked: boolean) =>
                checked ? setOpenPasswordModal(true) : setEnablePasswordProtection(checked)
              }
            />
          }
        />
        <SettingItem
          dataAid="hideAccountBalance"
          title={t('Security.hideAccountBalance')}
          endAndorment={
            <Switch
              dataAid="hideAccountBalanceSwitch"
              id="hide-account-balance"
              checked={isHideAccountBlalance}
              onChange={(checked: boolean) =>
                checked ? setHideAccountBlalance(true) : setHideAccountBlalance(checked)
              }
            />
          }
        />
        <SettingItem
          dataAid="autoLockNavigation"
          title={t('Security.autoLockTimer')}
          onClick={() => navigate('/settings/security/auto-lock-timer')}
          endAndorment={<ArrowRight />}
        />
        <SettingItem
          dataAid="passwordChangeNavigation"
          title={t('Security.changePassword')}
          endAndorment={<ArrowRight />}
          onClick={() => navigate('/settings/change-password')}
        />
      </div>

      <PasswordPromptModal
        modalState={openPasswordModal}
        closePromptModal={() => setOpenPasswordModal(false)}
        onSuccess={() => {
          setEnablePasswordProtection(true)
          setOpenPasswordModal(false)
        }}
        onFail={passwordPromptFailure}
      />
    </SinglePageTitleLayout>
  )
}

export default Security
