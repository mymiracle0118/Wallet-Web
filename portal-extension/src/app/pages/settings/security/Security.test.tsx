import React from 'react'
import Security from 'pages/settings/security/Security'
import AutoLockTimer from 'pages/settings/security/AutoLockTimer'
import { MemoryRouter as Router } from 'react-router-dom'
import { createMocks } from 'react-idle-timer'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { useWallet } from '@portal/shared/hooks/useWallet'
import { FormProvider, useForm } from 'react-hook-form'

const walletMock = {
  lockTime: '1',
  setLockTimer: () => (walletMock['lockTime'] = '5'),
}

const WrapperForm = ({ children }) => {
  const methods = useForm()
  return <FormProvider {...methods}>{children}</FormProvider>
}
jest.mock('@portal/shared/hooks/useWallet', () => ({
  useWallet: () => walletMock,
}))

describe('Security Page UI', () => {
  test('renders correctly Security Page', () => {
    const tree = render(<Security />)
    expect(tree).toMatchSnapshot()
  })
})

describe('Auto Lock Timer Page UI', () => {
  test('renders correctly Auto Lock Timer Page', () => {
    const tree = render(
      <WrapperForm>
        <AutoLockTimer />
      </WrapperForm>
    )
    expect(tree).toMatchSnapshot()
  })

  test('Test auto lock timer', async () => {
    const user = userEvent.setup()
    createMocks()
    render(
      <Router>
        <WrapperForm>
          <AutoLockTimer />
        </WrapperForm>
      </Router>
    )

    const newTime = '5'
    await user.type(screen.getByTestId('input-lock-time'), newTime)
    await user.click(screen.getByTestId('button-save'))
    const { lockTime } = useWallet()
    expect(lockTime).toBe(newTime)
  })
})
