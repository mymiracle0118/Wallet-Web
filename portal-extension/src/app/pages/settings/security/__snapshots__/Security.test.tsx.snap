// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auto Lock Timer Page UI renders correctly Auto Lock Timer Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dark:bg-surface-dark bg-custom-white flex flex-col h-[600px] dark:text-custom-white text-custom-black"
      >
        <div
          class="flex flex-row relative justify-between items-center h-16"
        >
          <button
            class="h-[24px] w-[24px] p-1 ml-1 hover:bg-custom-grey10 dark:hover:bg-custom-white10 rounded-full flex items-center justify-center absolute"
            type="button"
          >
            <arrow-left.svg
              classname="text-xl stroke-custom-black dark:stroke-custom-white"
            />
          </button>
          <div
            class="flex-1 text-center"
          >
            <div
              class="text-custom-black dark:text-custom-white text-lg font-extrabold"
              data-aid="autoLockHead"
            >
              Security.autoLockTimer
            </div>
          </div>
        </div>
        <div
          class="w-full h-full overflow-auto"
        >
          <div
            class="p-4"
          >
            <p
              class="text-sm text-custom-white pb-4"
            >
              Set the idle time 
              <span
                class="font-extrabold"
              >
                in minutes
              </span>
               before the wallet will be locked.
            </p>
            <div
              class="flex relative rounded-md items-center hover:shadow-md w-full bg-input-light-bg dark:bg-custom-white10"
            >
              <input
                autocomplete="off"
                class="outline-none	py-3.5 px-3.5 w-full text-sm text-custom-black dark:text-custom-white bg-transparent bg-transparent text-h4 "
                data-aid="inputTime"
                data-testid="input-lock-time"
                name="test"
                placeholder="1-120 minutes"
                type="number"
                value="1"
              />
            </div>
            <div
              class="flex mt-4 justify-center gap-2"
            >
              <button
                class="flex items-center justify-center rounded-md hover:cursor-pointer font-extrabold text-sm w-full h-[44px] border-[1px] border-solid text-primary dark:text-custom-white border-primary dark:border-custom-white40 dark:hover:border-custom-white hover:border-primary-alt hover:text-primary-alt"
                data-aid="cancelButton"
                type="button"
              >
                Actions.cancel
              </button>
              <button
                class="flex items-center justify-center rounded-md hover:cursor-pointer font-extrabold text-sm w-full h-[44px] text-custom-white dark:text-custom-white hover:bg-primary-alt dark:hover:bg-primary-alt bg-primary"
                data-aid="saveButton"
                data-testid="button-save"
                type="button"
              >
                Actions.save
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dark:bg-surface-dark bg-custom-white flex flex-col h-[600px] dark:text-custom-white text-custom-black"
    >
      <div
        class="flex flex-row relative justify-between items-center h-16"
      >
        <button
          class="h-[24px] w-[24px] p-1 ml-1 hover:bg-custom-grey10 dark:hover:bg-custom-white10 rounded-full flex items-center justify-center absolute"
          type="button"
        >
          <arrow-left.svg
            classname="text-xl stroke-custom-black dark:stroke-custom-white"
          />
        </button>
        <div
          class="flex-1 text-center"
        >
          <div
            class="text-custom-black dark:text-custom-white text-lg font-extrabold"
            data-aid="autoLockHead"
          >
            Security.autoLockTimer
          </div>
        </div>
      </div>
      <div
        class="w-full h-full overflow-auto"
      >
        <div
          class="p-4"
        >
          <p
            class="text-sm text-custom-white pb-4"
          >
            Set the idle time 
            <span
              class="font-extrabold"
            >
              in minutes
            </span>
             before the wallet will be locked.
          </p>
          <div
            class="flex relative rounded-md items-center hover:shadow-md w-full bg-input-light-bg dark:bg-custom-white10"
          >
            <input
              autocomplete="off"
              class="outline-none	py-3.5 px-3.5 w-full text-sm text-custom-black dark:text-custom-white bg-transparent bg-transparent text-h4 "
              data-aid="inputTime"
              data-testid="input-lock-time"
              name="test"
              placeholder="1-120 minutes"
              type="number"
              value="1"
            />
          </div>
          <div
            class="flex mt-4 justify-center gap-2"
          >
            <button
              class="flex items-center justify-center rounded-md hover:cursor-pointer font-extrabold text-sm w-full h-[44px] border-[1px] border-solid text-primary dark:text-custom-white border-primary dark:border-custom-white40 dark:hover:border-custom-white hover:border-primary-alt hover:text-primary-alt"
              data-aid="cancelButton"
              type="button"
            >
              Actions.cancel
            </button>
            <button
              class="flex items-center justify-center rounded-md hover:cursor-pointer font-extrabold text-sm w-full h-[44px] text-custom-white dark:text-custom-white hover:bg-primary-alt dark:hover:bg-primary-alt bg-primary"
              data-aid="saveButton"
              data-testid="button-save"
              type="button"
            >
              Actions.save
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Security Page UI renders correctly Security Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dark:bg-surface-dark bg-custom-white flex flex-col h-[600px] dark:text-custom-white text-custom-black"
      >
        <div
          class="flex flex-row relative justify-between items-center h-16"
        >
          <button
            class="h-[24px] w-[24px] p-1 ml-1 hover:bg-custom-grey10 dark:hover:bg-custom-white10 rounded-full flex items-center justify-center absolute"
            type="button"
          >
            <arrow-left.svg
              classname="text-xl stroke-custom-black dark:stroke-custom-white"
            />
          </button>
          <div
            class="flex-1 text-center"
          >
            <div
              class="text-custom-black dark:text-custom-white text-lg font-extrabold"
            >
              Security
            </div>
          </div>
        </div>
        <div
          class="w-full h-full overflow-auto"
        >
          <div
            class="MuiBox-root css-10l65i2"
          >
            <div
              style="width: 70%;"
            >
              <div
                class="text-custom-black dark:text-custom-white text-sm font-extrabold"
                data-aid="secretRecoveryPhraseNavigation"
              >
                Security.secretRecoveryPhrase
              </div>
            </div>
            <svg
              class="inline-block select-none shrink-0 fill-custom-black dark:fill-custom-white text-2xl"
              style="height: 1em; width: 1em;"
            >
              <arrow-right.svg />
            </svg>
          </div>
          <div
            class="MuiBox-root css-1mdx561"
          >
            <div
              style="width: 70%;"
            >
              <div
                class="text-custom-black dark:text-custom-white text-sm font-extrabold"
                data-aid="passwordProtectionPop"
              >
                Security.passwordProtectionTitle
              </div>
              <div
                class="text-custom-grey100 dark:text-custom-white40 text-sm font-regular"
                data-aid="passwordProtectionPop"
              >
                Security.passwordProtectionSubTitle
              </div>
            </div>
            <button
              aria-checked="false"
              class="!bg-custom-grey10 dark:!bg-custom-white40 relative inline-flex h-8 w-14 items-center rounded-[22px] transition-colors"
              data-aid="passwordSwitch"
              data-testid="switch-switch-enable-password-protection"
              id="headlessui-switch-:r0:"
              role="switch"
              tabindex="0"
              type="button"
            >
              <span
                class="translate-x-1 inline-block h-6 w-6 rounded-[18px] bg-custom-white transform transition ease-in-out duration-200 shadow-md"
              />
            </button>
          </div>
          <div
            class="MuiBox-root css-10l65i2"
          >
            <div
              style="width: 70%;"
            >
              <div
                class="text-custom-black dark:text-custom-white text-sm font-extrabold"
                data-aid="autoLockNavigation"
              >
                Security.autoLockTimer
              </div>
            </div>
            <svg
              class="inline-block select-none shrink-0 fill-custom-black dark:fill-custom-white text-2xl"
              style="height: 1em; width: 1em;"
            >
              <arrow-right.svg />
            </svg>
          </div>
          <div
            class="MuiBox-root css-10l65i2"
          >
            <div
              style="width: 70%;"
            >
              <div
                class="text-custom-black dark:text-custom-white text-sm font-extrabold"
                data-aid="passwordChangeNavigation"
              >
                Security.changePassword
              </div>
            </div>
            <svg
              class="inline-block select-none shrink-0 fill-custom-black dark:fill-custom-white text-2xl"
              style="height: 1em; width: 1em;"
            >
              <arrow-right.svg />
            </svg>
          </div>
          <div
            style="position: absolute; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
          />
        </div>
      </div>
    </div>
    <div
      id="headlessui-portal-root"
    >
      <div
        data-headlessui-portal=""
      >
        <div />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dark:bg-surface-dark bg-custom-white flex flex-col h-[600px] dark:text-custom-white text-custom-black"
    >
      <div
        class="flex flex-row relative justify-between items-center h-16"
      >
        <button
          class="h-[24px] w-[24px] p-1 ml-1 hover:bg-custom-grey10 dark:hover:bg-custom-white10 rounded-full flex items-center justify-center absolute"
          type="button"
        >
          <arrow-left.svg
            classname="text-xl stroke-custom-black dark:stroke-custom-white"
          />
        </button>
        <div
          class="flex-1 text-center"
        >
          <div
            class="text-custom-black dark:text-custom-white text-lg font-extrabold"
          >
            Security
          </div>
        </div>
      </div>
      <div
        class="w-full h-full overflow-auto"
      >
        <div
          class="MuiBox-root css-10l65i2"
        >
          <div
            style="width: 70%;"
          >
            <div
              class="text-custom-black dark:text-custom-white text-sm font-extrabold"
              data-aid="secretRecoveryPhraseNavigation"
            >
              Security.secretRecoveryPhrase
            </div>
          </div>
          <svg
            class="inline-block select-none shrink-0 fill-custom-black dark:fill-custom-white text-2xl"
            style="height: 1em; width: 1em;"
          >
            <arrow-right.svg />
          </svg>
        </div>
        <div
          class="MuiBox-root css-1mdx561"
        >
          <div
            style="width: 70%;"
          >
            <div
              class="text-custom-black dark:text-custom-white text-sm font-extrabold"
              data-aid="passwordProtectionPop"
            >
              Security.passwordProtectionTitle
            </div>
            <div
              class="text-custom-grey100 dark:text-custom-white40 text-sm font-regular"
              data-aid="passwordProtectionPop"
            >
              Security.passwordProtectionSubTitle
            </div>
          </div>
          <button
            aria-checked="false"
            class="!bg-custom-grey10 dark:!bg-custom-white40 relative inline-flex h-8 w-14 items-center rounded-[22px] transition-colors"
            data-aid="passwordSwitch"
            data-testid="switch-switch-enable-password-protection"
            id="headlessui-switch-:r0:"
            role="switch"
            tabindex="0"
            type="button"
          >
            <span
              class="translate-x-1 inline-block h-6 w-6 rounded-[18px] bg-custom-white transform transition ease-in-out duration-200 shadow-md"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-10l65i2"
        >
          <div
            style="width: 70%;"
          >
            <div
              class="text-custom-black dark:text-custom-white text-sm font-extrabold"
              data-aid="autoLockNavigation"
            >
              Security.autoLockTimer
            </div>
          </div>
          <svg
            class="inline-block select-none shrink-0 fill-custom-black dark:fill-custom-white text-2xl"
            style="height: 1em; width: 1em;"
          >
            <arrow-right.svg />
          </svg>
        </div>
        <div
          class="MuiBox-root css-10l65i2"
        >
          <div
            style="width: 70%;"
          >
            <div
              class="text-custom-black dark:text-custom-white text-sm font-extrabold"
              data-aid="passwordChangeNavigation"
            >
              Security.changePassword
            </div>
          </div>
          <svg
            class="inline-block select-none shrink-0 fill-custom-black dark:fill-custom-white text-2xl"
            style="height: 1em; width: 1em;"
          >
            <arrow-right.svg />
          </svg>
        </div>
        <div
          style="position: absolute; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
